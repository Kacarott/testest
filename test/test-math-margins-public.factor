! Copyright 2022 nomennescio

USING: accessors debugger io kernel math.constants math.margins namespaces present sequences tools.testest ;
IN: tests

: run-tests ( -- )
  "math.margins" describe#{
    "constructors" it#{
      <{ 1 2 3 <margin> not -> f }>
      <{ 2 1 [a-e,a+e] not -> f }>
      <{ 2 50 [a-%,a+%] not -> f }>
    }#
    "constructors aliases" it#{
      <{ 2 1 ± not -> f }>
      <{ 2 50 ±% not -> f }>
    }#
    "conversion" it#{
      <{ "no" >margin -> f }>
      <{ 2 >± -> 2 0 ± }>
      <{ 2 0 ± >± -> 2 0 ± }>
    }#
    "equals" it#{
      <{ 2 >margin 2 2 2 <margin> = -> t }>
      <{ 2 >margin 2 1 [a-e,a+e] = -> t }>
      <{ 3 >margin pi 5 [a-%,a+%] = -> t }>
    }#
  }#
;

MAIN: run-tests
