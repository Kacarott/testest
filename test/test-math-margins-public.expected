
<DESCRIBE::>math.margins

<IT::>constructors

<PASSED::>Test Passed

<PASSED::>Test Passed

<COMPLETEDIN::>0.076738 ms

<IT::>constructors aliases

<PASSED::>Test Passed

<PASSED::>Test Passed

<COMPLETEDIN::>0.012251 ms

<IT::>conversion

<PASSED::>Test Passed

<PASSED::>Test Passed

<PASSED::>Test Passed

<COMPLETEDIN::>0.084875 ms

<IT::>equals

<PASSED::>Test Passed

<PASSED::>Test Passed

<COMPLETEDIN::>0.028185 ms

<COMPLETEDIN::>0.383209 ms
